foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,67 @@
+import {EventEmitter} from 'events';
+
+import Authorizer from './Authorizer';
+import {CHANGE} from './AuthEvent';
+
+/**
+ * List of authorizers
+ *
+ * @type {Array.<Authorizer>}
+ */
+const authorizers = [];
+
+/**
+ * AuthService
+ */
+class AuthService extends EventEmitter {
+
+  /**
+   * Register authorizer
+   *
+   * @param {Authorizer} authorizer
+   */
+  registerAuthorizer(authorizer) {
+    if (!authorizer || !(authorizer instanceof Authorizer)) {
```

Comment:
It ensures the all  authorizer are objects with a method named `authorize` which will return a boolean.